### ISSUE TITLE
Segfault for specific case

### ISSUE THREAD

**Issue by hilmarm** [sweeneychris/RpolyPlusPlus#2]
Creating the 6th degree polynomials with coefficients as given at the end
of this message ends with a segmentation fault error. I don't remember
which of the 4 cases given crashes, but I think it was number 2 or 3.
Running RPolyPlusPlus with the commit from Aug 11, 2015 will
not result in the crash. This means the error only happens with the
newest commit (on Mar 9, 2016).

Hilmar

coeffs = 
           1
     -359951
 3.78335e+10
-1.37262e+15
 5.26491e+18
           0
           0

coeffs = 
           1
     -611771
 8.23017e+10
-5.62821e+14
-4.18369e+18
           0
           0

coeffs = 
          1
    -691197
1.27678e+11
-6.9045e+15
8.80928e+19
          0
          0

coeffs = 
           1
     -280525
 2.53443e+10
-7.45354e+14
-7.27967e+17
           0
           0

**Comment by sweeneychris** [comment]
Thanks for reporting this. It has to do with the current convegence criterion for the algorithm. If I look at the roots the that linear and quadratic shifts are finding I notice that they are indeed correct. But the algorithm is not properly detecting when to stop.

I'm using the initial recommendation from Jenkin's Traub, though there are surely more accurate methods available. I'm reading the relevant literature now and will submit a fix soon.

**Comment by sweeneychris** [comment]
I have uploaded a patch to the master branch that uses a much more sound convergence criterion. Additionally, I tested each of the 4 polynomials you provided and verified that they produce correct results with no more segfaults.

I may need to adjust the thresholds [here](https://github.com/sweeneychris/RpolyPlusPlus/blob/master/src/find_polynomial_roots_jenkins_traub.cc#L170) so let me know if you have any issues using this in practice.
